apt/private/lockfile.bzl

```diff
@@ -16,11 +16,14 @@ def _add_package(lock, package, arch):
     k = _package_key(package, arch)
     if k in lock.fast_package_lookup:
         return
+    filename = package["Filename"]
+    if filename.startswith("./"):
```

Comment:
When `Packages` lives in the same directory/URL that the packages themselves live, the manifest has `./` as the leading file path for all the packages listed, however we don't want a `./` in the URL.

Reply:
FYI, this is not what the spec says, check https://wiki.debian.org/DebianRepository/Format#Filename:

> Filename
> 
> The mandatory Filename field shall list the path of the package archive relative to the base directory of the repository. The path should be in canonical form, that is, without any components denoting the current or parent directory ("." or ".."). It also should not make use of any protocol-specific components, such as URL-encoded parameters.

I've checked the R-Project flat repos and this doesn't happen / is not needed. However, this is something that happens in the NVIDIA CUDA flat repos . Thus, I think that this it's a quirk of some specific flat repos that don't conform with the Debian repo spec.

Reply:
Super interesting. I was looking for documentation on the manifest format for a debian repository and was having a hard time finding anything authoritative. Thanks for sharing the link.